include/MiniLua/values.hpp

```diff
@@ -664,6 +664,23 @@ auto operator==(const UnaryOrigin&, const UnaryOrigin&) noexcept -> bool;
 auto operator!=(const UnaryOrigin&, const UnaryOrigin&) noexcept -> bool;
 auto operator<<(std::ostream&, const UnaryOrigin&) -> std::ostream&;
 
+/**
+ * Origin for a Value that was created in a nary operation (a function that receives n arguments)
+ * (or some functions with one argument) using val.
+ */
+struct NaryOrigin {
```

Comment:
If we stick with the name we should capitalize the a. I.e. `NAryOrigin`.

However I would prefer to use another name. While *binary* and *unary* should be easy to understand, *n-ary* is probably harder to understand. Also binary and unary was originally derived from *unary/binary operator*.

IMO something like `VariadicOrigin` is probably a better name because *variadic* should already be somewhat understood and there is the concept of *variadic arguments* (or varargs) for functions. I'm also open to other suggestions e.g. `MultiArgumentOrigin` would also be fine (and maybe even more clear).

Reply:
I just used this name to stick to the pattern of unary, binary ff for n arguments. 
But I agree that a more clear name would be better. I will change that.